src/linear_algebra/quadratic_form.lean

```diff
@@ -816,3 +817,128 @@ theorem exists_orthogonal_basis [hK : invertible (2 : K)]
 let ⟨v, hv₁, hv₂, _⟩ := exists_orthogonal_basis' hB₁ hB₂ in ⟨v, hv₁, hv₂⟩
 
 end bilin_form
+
+namespace quadratic_form
+
+open_locale big_operators
+
+open finset bilin_form
+
+variables {M₁ : Type*} [add_comm_group M₁] [module R M₁]
+variables {ι : Type*} [fintype ι] {v : ι → M}
+
+/-- A quadratic form composed with a `linear_equiv` is isometric to itself. -/
+def isometry_of_comp_linear_equiv (Q : quadratic_form R M) (f : M₁ ≃ₗ[R] M) :
+  Q.isometry (Q.comp (f : M₁ →ₗ[R] M)) :=
+{ map_app' :=
+  begin
+    intro,
+    simp only [comp_apply, linear_equiv.coe_coe, linear_equiv.to_fun_eq_coe,
+               linear_equiv.apply_symm_apply, f.apply_symm_apply],
+  end, .. f.symm }
+
+/-- Given a quadratic form `Q` and a basis, `of_is_basis` is the basis representation of `Q`. -/
+noncomputable def of_is_basis (Q : quadratic_form R M)
+  (hv₁ : is_basis R v) : quadratic_form R (ι → R) := Q.comp hv₁.equiv_fun.symm
```

Comment:
```suggestion
noncomputable def of_is_basis (Q : quadratic_form R M) (hv₁ : is_basis R v) :
  quadratic_form R (ι → R) :=
Q.comp hv₁.equiv_fun.symm
```

Reply:
Should this be called something like `basis_repr`?

Reply:
I like the name `basis_repr`